enhancements/subscription-content/subscription-injection.md

```diff
@@ -0,0 +1,487 @@
+
+1. `Bundle` supporting a single bundle injected into a workload within the same namespace, allowing
+   one entitlement secret and one yum.repo configuration.
+2. `Bundle` supporting multiple `entitlements` and `yumRepositories`.
+3. Allow multiple `Bundles` to be injected into a workload within the same namespace.
+4. `ClusterBundle` supporting a single bundle injected into a workload, allowing one entitlement
+   secret and one yum.repo configuration.
+4. `ClusterBundle` supporting multiple multiple `entitlements` and `yumRepositories`.
+5. Allow multiple `ClusterBundles` to be injected into a workload.
+6. Allow auto-injection of `ClusterBundles` into all workloads. 
+
+#### Dev Preview
+
+- Bundle APIs at `v1alpha1`.
+- Dev preview release of the projected resource CSI driver via OLM (as community operator)
+- Release as a community OLM operator
+
+##### Dev Preview -> Tech Preview
+
+- Projected resource CSI driver reaches tech preview state.
+- APIs reach `v1beta1`
+- Clear documentation on how to enable subscription injection across the cluster
+- Initial scale testing
+- Gather feedback for namespace-restricted subscription bundles
+- Release as a tech preview, official OLM operator
+
+##### Tech Preview -> GA
+
+- All APIs reach `v1` stability
+- Projected resource CSI driver reaches GA
+- Security audit for these components and the projected resource CSI driver
+- Scale testing (with potential mitigations like `HorizontalPodAutoscalers`)
+- Disruption testing (with potential mitigations like `PodDisruptionBudgets`)
+
+### Upgrade / Downgrade Strategy
+
+Upgrades and downgrades will be handled via OLM. Each released version will specify:
+
+- Minimum supported version of the projected resource CSI driver
+- Minimum supported kube version of the cluster.
+
+### Removal Strategy
+
+As an OLM operator, removing the Subscription Injection operator will not remove the associated
+custom resource definitions. As a result, existing `Bundle` and `ClusterBundle` objects will remain
+on the cluster and the associated injected volumes will remain intact.
+
+### Version Skew Strategy
+
+How will the component handle version skew with other components? What are the guarantees? Make
+sure this is in the test plan.
+
+Consider the following in developing a version skew strategy for this enhancement:
+
+- During an upgrade, we will always have skew among components, how will this impact your work?
+- Does this enhancement involve coordinating behavior in the control plane and in the kubelet? How
+  does an n-2 kubelet without this feature available behave when this feature is used?
+- Will any other components on the node change? For example, changes to CSI, CRI or CNI may require
+  updating that component before the kubelet.
+
+### Open Questions
+
+1. Is allowing an admission webhook which can read all `Secrets` and `ConfigMaps` acceptable?
+2. Do we need to perform a subject access review if a service account needs to read a `Bundle`
```

Comment:
so if the SA can't read the Bundle, the "escalation" here is effectively that someone can create a pod that references the bundle (even though they can't read it), which, by virtue of the mutating webhook, exposes the name of the secrets the bundle references(as mounts on the pod).  (The SA may or may not actually be able to read those secrets, so the pod may or may not ultimately be admitted).

it also means the user can probe at the names of Bundles which may exist in the namespace.

Overall this doesn't seem like an area of concern:
1) it's restricted within a namespace
2) it's users who can already create pods in the namespace, which means they are pretty privileged

Assuming i've defined the boundaries of the exploitability/escalation correctly, anyway.